lineman/app/components/group_form/group_form.coffee

```diff
@@ -0,0 +1,65 @@
+angular.module('loomioApp').factory 'GroupForm', ->
+  templateUrl: 'generated/components/group_form/group_form.html'
+  controller: ($scope, $rootScope, $location, group, FormService, Records, $translate, PrivacyString) ->
+    $scope.group = group.clone()
+
+    $scope.$on 'modal.closing', (event) ->
+      FormService.confirmDiscardChanges(event, $scope.group)
+
+    $scope.i18n = do ->
+      h = {}
```

Comment:
What is 'h'? And I kinda find that these deep if / else logic structures are really difficult to follow; I'd much rather have some sort of separate service or method that would determine each value on it's own, based on the group

```
.group-heading{translate: '{{groupHeadingFor(group)}}'}
.group-name{translate: '{{groupNameFor(group)}}'}
.group-submit{translate: '{{groupSubmitFor(group)}}'}
```